django/utils/http.py

```diff
@@ -176,10 +176,16 @@ def parse_http_date(date):
     try:
         year = int(m.group('year'))
         if year < 100:
-            if year < 70:
-                year += 2000
-            else:
-                year += 1900
+            # two-digit RFC850-date format year to be
+            # parsed relative to current year in accordance with RFC7231
+            current_year = datetime.datetime.utcnow().year
+            current_century = current_year - (current_year % 100)
+            assumed_year = current_century + year
+            if assumed_year - current_year > 50:
+                # years >50 in future are in the past
+                assumed_year -= 100
+            year = assumed_year
```

Comment:
Both `current_year` and `assumed_year` contains `current_century`, hence I think we can simplify this implementation, e.g. (IMO it's also more readable)
```python
current_year = datetime.datetime.utcnow().year
current_century = current_year - (current_year % 100)
if year - (current_year % 100) > 50:
    # year that appears to be more then 50 years in the future are
    # interpreted as representing the past.
    year += current_century - 100
else:
    year += current_century
```

Reply:
Thank you for the feedback! Agree this is also more readable.